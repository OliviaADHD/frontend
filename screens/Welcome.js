import React, {useState, useEffect} from "react";
import { StatusBar } from "expo-status-bar";
import { ActivityIndicator } from "react-native";
import {
    StyledContainer,
    InnerContainer,
    PageLogoWelcome,
    LoadScreenBottomImage,
    ExtraText,
}from './../components/styles';

const Welcome =(props) => {
    const [authLoaded, setAuthLoaded] = useState(false);

    useEffect(() => {
      const timer = setTimeout(() => {
        setAuthLoaded(true);
      }, 3000);
      return () => clearTimeout(timer); // To prevent leaks
    }, []);

    useEffect(() => {
      if (authLoaded) {
        props.navigation.replace('Login');
      }
    }, [authLoaded, props.navigation]);
    return (
        <StyledContainer>
            <StatusBar style="dark"/>
            <InnerContainer>
                <PageLogoWelcome source={require('./../assets/images/logo.png')} />
                <ExtraText>Helps You Stay Organized</ExtraText>
                <ActivityIndicator size="large" color="#694398"/>
                <LoadScreenBottomImage source={require('./../assets/images/loadScreenBottom.png')} />
            </InnerContainer>
        </StyledContainer>

    );
}


export default Welcome;