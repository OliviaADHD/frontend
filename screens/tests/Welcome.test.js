import React from 'react';
import {render} from '@testing-library/react-native';
import Welcome from '../Welcome';

import TestRenderer from 'react-test-renderer';

const{act} = TestRenderer;

const mockNavigation = {
    navigate: jest.fn(),
};
const TestProps ={
    navigation: mockNavigation,
};



test('Snapshot changed?', async () =>{
    const {debug} = render(<Welcome props={mockNavigation}/>);
    expect(debug).toMatchSnapshot();
});

/*
test('Went to login Screen after 3 sec', async () =>{
    const {debug} = render(<Welcome props={mockNavigation}/>);
    console.log("name of debug " + debug.name)
    const value = await new Promise(() => {
        setTimeout(()=>{console.log("name of debug "+debug.name)},
                        4000);
        return true;
                    });
    expect(value).toBe(true);
    
})
*/
