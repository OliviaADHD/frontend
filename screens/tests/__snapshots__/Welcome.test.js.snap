// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot changed? 1`] = `[Function]`;
